### PR TITLE
Use null instead of default, to avoid stomping on netmask

### LINKED ISSUE #23347: Netmask is overwritten with incorrect data
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

The /config/network page allows setting of static IP addresses. When a user enters an IP with a prefix length, it automatically updates the netmask to match the prefix. However, when a prefix length is not included (the default), it replaces the netmask with 255.255.255.255, and it's not obvious that this has happened. A user who does this will probably find the change applies but they're unable to reach HA from the network afterwards.

### Describe the behavior you expected

The netmask should only be changed if the user supplied a prefix length. A user who only changed the last digit of an IP address would not expect to check the netmask afterwards. It won't work to only change the netmask if not already set, because 255.255.255.255 isn't a helpful default.

The prefix length is also not populated in the other direction - editing the mask to 255.255.255.0 does not append /24 to the IP address.

### Steps to reproduce the issue

1. Visit /config/network
2. Start configuring a Static IPv4 for one of the interfaces (you don't need to save the change)
3. Set the netmask to 255.255.255.0 (this may already be the case)
4. Set the IP address to 192.168.1.1 and click or tab out of the field
5. The netmask is replaced with 255.255.255.255, which is incorrect

### What version of Home Assistant Core has the issue?

2024.12.4

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Chrome 131.0.6778.109

### Which operating system are you using to run this browser?

Windows 11

### State of relevant entities

```txt
N/A
```

### Problem-relevant frontend configuration

```yaml
You probably already have a network adapter with IPv4.
```

### Javascript errors shown in your browser console/inspector

```txt
N/A
```

### Additional information

- `_handleInputValueChanged` calls `parseAddress` [when the address is changed](https://github.com/home-assistant/frontend/blob/2fe6203/src/panels/config/network/supervisor-network.ts#L658)
- `parseAddress` assumes the IP address is in CIDR notation and calls `cidrToNetmask` [on the prefix length](https://github.com/home-assistant/frontend/blob/2fe6203/src/data/hassio/network.ts#L120-L121)
- `cidrToNetmask`  calculates an undefined prefix as `NaN`, resulting in a `bits` of -1, which is the same as /32
- `_handleInputValueChanged` updates the netmask [to the result](https://github.com/home-assistant/frontend/blob/2fe6203/src/panels/config/network/supervisor-network.ts#L659-L660)

### PR DESCRIPTION
**Pull request by marksteward**
## Proposed change

Refines https://github.com/home-assistant/frontend/pull/23348 to return `null` instead of a default netmask. This then allows the code to avoid stomping on a preexisting netmask that isn't /24 for IPv4 or /64 for IPv6.

To be clear, the issue was never that the default was wrong, but that the netmask is overwritten as soon as the IP address field blurs, and the previous PR doesn't change that. A user tweaking the last octet of their IP address and clicking Save will not have the opportunity to notice the changed netmask before it applies and they lose access. /24 is less likely to break things but people do use /25s and 10.x.x.x networks. And /64 is the smallest used subnet for IPv6, /48 is quite often given out by ISPs.

This PR also switches to checking `ip` for colons instead of the whole `address` field, but that's very minor as nothing's validated at this stage.

## Type of change
- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
N/A

## Additional information

- This PR fixes or closes issue: fixes #23347, #23246
- This PR is related to issue or discussion: #23348
- Link to documentation pull request:

## Checklist
- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/network/supervisor-network.ts [hunk 1]
```diff
@@ -656,8 +656,10 @@ export class HassioNetwork extends LitElement {
     if (id === "address") {
       const index = (ev.target as any).index as number;
       const { mask } = parseAddress(value);
-      this._interface[version]!.address![index] = formatAddress(value, mask);
-      this.requestUpdate("_interface");
+      if (mask) {
```
Comment (bramkragten) [inline src/panels/config/network/supervisor-network.ts:659]: This means a change in IP address is never picked up as long as there is no mask?

Also means that if there is a re-render before the mask is updated, the changes of the user are undone.

Comment (marksteward) [reply src/panels/config/network/supervisor-network.ts]: Ahh yep, just realised this after playing some more. It looks like adding an empty IP address adds with /24 or /64, so there's always a netmask behind the scenes, which we can use as the default instead.

**Review by bramkragten — DISMISSED** [review]

**Comment by marksteward** [comment]
(I've run `yarn run lint:types` locally now)

**Comment by silamon** [comment]
> (I've run `yarn run lint:types` locally now)

To get CI green, you may need to run prettier as well 

`yarn run format:prettier`

**Review by bramkragten — APPROVED** [review]

**Pull request merged**